src/JSInterop/Microsoft.JSInterop/src/IJSRuntime.cs

```diff
@@ -13,20 +14,10 @@ public interface IJSRuntime
         /// <summary>
         /// Invokes the specified JavaScript function asynchronously.
         /// </summary>
-        /// <typeparam name="T">The JSON-serializable return type.</typeparam>
+        /// <typeparam name="TValue">The JSON-serializable return type.</typeparam>
         /// <param name="identifier">An identifier for the function to invoke. For example, the value <code>"someScope.someFunction"</code> will invoke the function <code>window.someScope.someFunction</code>.</param>
         /// <param name="args">JSON-serializable arguments.</param>
-        /// <returns>An instance of <typeparamref name="T"/> obtained by JSON-deserializing the return value.</returns>
-        Task<T> InvokeAsync<T>(string identifier, params object[] args);
-
-        /// <summary>
-        /// Stops tracking the .NET object represented by the <see cref="DotNetObjectRef"/>.
-        /// This allows it to be garbage collected (if nothing else holds a reference to it)
-        /// and means the JS-side code can no longer invoke methods on the instance or pass
-        /// it as an argument to subsequent calls.
-        /// </summary>
-        /// <param name="dotNetObjectRef">The reference to stop tracking.</param>
-        /// <remarks>This method is called automatically by <see cref="DotNetObjectRef.Dispose"/>.</remarks>
-        void UntrackObjectRef(DotNetObjectRef dotNetObjectRef);
```

Comment:
I removed this API. We recommendation would be to just dispose the instance since it would do exactly the same. Is there a good reason to keep this?

Reply:
This is interesting. Previously, it was the JS side that disposed the object references. The main reason for that was that when calling back to .NET, the .NET side would just receive the wrapped object instance, not the wrapper itself, so it had no easy way to dispose the wrapper.

With the new design, the .NET side does have an easy opportunity to dispose wrappers when handling any callback, so I definitely agree it's good for .NET to be able to dispose from its own side like you've added now.

I'm not certain what scenarios remain for disposing from the JS side. Maybe there are none. Even if there are, it could be achieved by calling a method on .NET to ask it to do the disposal. So unless anyone comes up with objects, it think it's good to proceed with your updated design where disposal only happens from the .NET side.

Reply:
We still have a way to dispose the type from JavaScript - however it's an internal implementation detail. Calling `dotNetRef.Dispose()` works exactly the way it works today, it calls a JSInvokable method that passes in the object id. We look it up and dispose it. This seemed superfluous. Given that there isn't a way for `IJsRuntime` to track or create a `DotNetObjectRef`, this seemed unnecessary.

Reply:
OK, now I understand how the disposal-from-JS is done.

I totally agree with all your reasoning here!